File: epi/Ruby/13.01.epi.rb
=begin
Funtion takes as input a set of words and prints groups a anagrams for those word
=end
require 'set'

def anagrams(set)
	hash = Hash.new
	set.each do |str|
		sorted_one = str.chars.sort.join
		hash[sorted_one] = Array.new if hash[sorted_one].nil?
		hash[sorted_one] << str
	end

	hash.each do |key, value|
		if value.length >= 2
			to_print = ""
			value.each do |word|
				to_print << word + ", "
			end
			puts to_print.strip
		end
	end
end

set = Set.new(['hahaha', 'ahahah', 'hoho', 'ohoh', 'hello', 'hey'])
anagrams(set)

File: epi/Javascript/13.01.epi.js
/*
Write a function that takes as input a set of words and prints groups of anagrams for
those words
*/
function groupsAnagrams(arr) {
    if (arr.length == 0) {
        return;
    }
    var hash = {};
    for (var i=0; i<arr.length; i++) {
        var s = arr[i].split("").sort().join("");
        if (hash[s] == undefined) {
            hash[s] = [];
        }
        hash[s].push(arr[i]);
    }
    for (s in hash) {
        if (hash[s].length > 1) {
            console.log(hash[s]);
        }
    }
}
var arr = ["abc", "cba", "abd", "dab", "bac", "abe"];
groupsAnagrams(arr);
